### PR TITLE
chore(dep)!: upgrade gts 2.0.0

### PR DESCRIPTION
**Pull request by summer-ji-eng**
upgrade gts 2.0.0
fix some lint issue.

### CODE REVIEW

File: test/index.ts [hunk 1]
```diff
@@ -179,7 +185,7 @@ describe('paginator', () => {
 
     it('should apply the proper context', done => {
       const parsedArguments = {a: 'b', c: 'd'} as ParsedArguments;
-      FakeClass.prototype.methodToExtend = function() {
+      FakeClass.prototype.methodToExtend = function () { // eslint-disable-line
```
Comment (summer-ji-eng) [inline test/index.ts]: got linter 
`error    Delete `·`                                prettier/prettier`

File: test/resource-stream.ts [hunk 2]
```diff
@@ -31,12 +31,12 @@ describe('ResourceStream', () => {
   let requestSpy: sinon.SinonSpy;
   let stream: ResourceStream<{}>;
 
-  beforeEach(() => {
+  beforeEach(() => { // eslint-disable-line
```
Comment (summer-ji-eng) [inline test/resource-stream.ts]: got linter error
`'beforeEach' is not defined                     no-undef`

File: test/index.ts [hunk 3]
```diff
@@ -22,14 +22,16 @@ import * as P from '../src';
...
+    // do nothing
+  },
 };
 
 class FakeResourceStream extends Transform {
   calledWith: IArguments;
   constructor() {
     super({objectMode: true});
-    this.calledWith = arguments;
+    this.calledWith = arguments; // eslint-disable-line
```
Comment (summer-ji-eng) [inline test/index.ts]: got linter error
`Use the rest parameters instead of 'arguments'  prefer-rest-params`

File: package.json [hunk 4]
```diff
@@ -36,11 +36,15 @@
     "@types/proxyquire": "^1.3.28",
     "@types/sinon": "^7.0.0",
     "@types/uuid": "^7.0.0",
+    "c8": "^7.0.0",
     "codecov": "^3.0.4",
-    "gts": "^1.0.0",
+    "eslint": "^6.8.0",
```
Comment (JustinBeckwith) [inline package.json]: We can drop all of these `eslint` dependencies

File: test/index.ts [hunk 5]
```diff
@@ -39,19 +42,23 @@ const p = proxyquire('../src', {
 
 const sandbox = sinon.createSandbox();
 
+// eslint-disable-next-line no-undef
```
Comment (JustinBeckwith) [inline test/index.ts]: To fix this one, make sure `afterEach` is imported as part of the mocha requires.

```js
import {describe, it, afterEach} from 'mocha';
```

Then you should be able to get rid of these exceptions

File: test/index.ts [hunk 6]
```diff
@@ -39,19 +42,23 @@ const p = proxyquire('../src', {
...
 }
 
 describe('paginator', () => {
   const UUID = uuid.v1();
-  function FakeClass() {}
+  function FakeClass() {
+    // do nothing
+  }
 
+  /* eslint-disable-next-line no-undef */
```
Comment (JustinBeckwith) [inline test/index.ts]: Same as above for `afterEach`

File: test/index.ts [hunk 7]
```diff
@@ -75,7 +82,9 @@ describe('paginator', () => {
 
     it('should accept an array or string method names', () => {
       const originalMethod = FakeClass.prototype.methodToExtend;
-      FakeClass.prototype.anotherMethodToExtend = () => {};
+      FakeClass.prototype.anotherMethodToExtend = () => {
+        // do nothing
```
Comment (JustinBeckwith) [inline test/index.ts]: @bcoe @alexander-fenster I think this rule is annoying and we should get rid of it.  Thoughts?

File: tsconfig.json [hunk 8]
```diff
@@ -1,6 +1,7 @@
 {
   "extends": "./node_modules/gts/tsconfig-google.json",
   "compilerOptions": {
+    "lib": ["es2015", "dom"],
```
Comment (JustinBeckwith) [inline tsconfig.json]: We should be switching these to `es2018` to snap with the latest node.js support via https://node.green

File: package.json [hunk 9]
```diff
@@ -7,16 +7,16 @@
...
     "fix": "gts fix",
+    "prelint": "cd samples; npm link ../; npm i",
+    "lint": "gts check",
     "prepare": "npm run compile",
     "pretest": "npm run compile",
     "posttest": "npm run lint",
     "docs": "compodoc src/",
     "presystem-test": "npm run compile",
-    "samples-test": "cd samples/ && npm link ../ && npm test && cd ../",
+    "samples-test": "cd samples/ && npm link ../ && npm install && npm test && cd ../",
```
Comment (JustinBeckwith) [inline package.json]: @bcoe do we do this in other places?

Comment (bcoe) [reply package.json]: all that should be required is this:

```
"samples-test": "cd samples/ && npm link ../ && npm test && cd ../"
```

which is what we use in other libraries, `npm link ../` should perform the install.

File: test/index.ts (hunk 5 above)
Comment (summer-ji-eng) [reply test/index.ts]: thanks

File: test/resource-stream.ts [hunk 10]
```diff
@@ -31,11 +31,13 @@ describe('ResourceStream', () => {
   let requestSpy: sinon.SinonSpy;
   let stream: ResourceStream<{}>;
 
+  // eslint-disable-next-line no-undef
   beforeEach(() => {
     requestSpy = sandbox.spy();
     stream = new ResourceStream(config, requestSpy);
   });
 
+  // eslint-disable-next-line no-undef
```
[the remark this answers is not in the document]
Comment (summer-ji-eng) [reply test/resource-stream.ts]: did, thanks

File: package.json (hunk 9 above)
Comment (summer-ji-eng) [reply package.json]: FYI: The couple libraries I converted use `samples-test": "cd samples/ && npm link ../ && npm install && npm test && cd ../`, like https://github.com/googleapis/nodejs-monitoring/blob/master/package.json#L32

File: src/resource-stream.ts [hunk 11]
```diff
@@ -47,7 +47,7 @@ export class ResourceStream<T> extends Transform implements ResourceEvents<T> {
     this._requestsMade = 0;
     this._resultsToSend = args.maxResults === -1 ? Infinity : args.maxResults!;
   }
-  // tslint:disable-next-line:no-any
+  /* eslint-disable  @typescript-eslint/no-explicit-any */
```
Comment (bcoe) [inline src/resource-stream.ts]: I think this is fine for now, I would rather that these PRs turn into as minimal of a refactor as possible, even though we might want to come back and make this a more extensive refactor.

**Pull request merged**